compiler/rustc_ast_lowering/src/asm.rs

```diff
@@ -199,6 +199,19 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                     }
                 );
 
+                // Some register classes can only be used as clobbers.
+                if reg_class.is_clobber_only(asm_arch.unwrap())
+                    && (!is_clobber || matches!(reg, asm::InlineAsmRegOrRegClass::RegClass(_)))
```

Comment:
I'm not following what purpose the `matches!(reg, asm::InlineAsmRegOrRegClass::RegClass(_))` part is serving. Could you explain and/or add a comment?